core/embed/bootloader/main.c

```diff
@@ -541,15 +549,16 @@ int bootloader_main(void) {
   uint32_t touched = 0;
 #ifdef USE_TOUCH
   if (firmware_present == sectrue && stay_in_bootloader != sectrue) {
-    for (int i = 0; i < 100; i++) {
+    touch_wait_until_ready();
+    for (int i = 0; i < 10; i++) {
       touched = touch_is_detected() | touch_read();
       if (touched) {
         break;
       }
 #ifdef TREZOR_EMULATOR
       hal_delay(25);
 #else
-      hal_delay(1);
+      hal_delay_us(1000);
```

Comment:
why?

Reply:
The hal_delay function does delay _at least_ that amount of milliseconds. In reality, it waits for two n+1 systicks, as the time needed to reach the first systick is unknown. But when done in loop, the first systick (on second and next pass throughs) will be almost the whole millisecond, therefore the total delay will be n+1 ms.  Resulting here in double of the intended delay.